File: app/src/main/java/com/sami/rippel/ui/adapter/DetailPagerAdapter.java
package com.sami.rippel.ui.adapter;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.RelativeLayout;

import com.bumptech.glide.request.target.SimpleTarget;
import com.bumptech.glide.request.transition.Transition;
import com.sami.rippel.allah.R;
import com.sami.rippel.base.BasePagerAdapter;
import com.sami.rippel.base.BaseView;
import com.sami.rippel.model.ViewModel;
import com.sami.rippel.model.entity.WallpaperObject;
import com.sami.rippel.ui.activity.DetailsActivity;
import com.sami.rippel.views.GlideApp;
import com.sami.rippel.views.TouchImageView;

import java.util.ArrayList;

public class DetailPagerAdapter extends BasePagerAdapter<WallpaperObject> {
    private Context mContext;
    private int mResourceId;
    private BaseView baseView;

    public DetailPagerAdapter(Context context, int resourceId,
                              ArrayList<WallpaperObject> data, BaseView baseView) {
        super(data);
        this.mContext = context;
        this.mResourceId = resourceId;
        this.baseView = baseView;
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        final WallpaperObject item = getItem(position);
        LayoutInflater inflater = (LayoutInflater) mContext
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View viewLayout = inflater.inflate(mResourceId, container, false);
        final TouchImageView mDetailImage = (TouchImageView) viewLayout
                .findViewById(R.id.detailImage);
        TouchImageView img = new TouchImageView(container.getContext());
        baseView.showLoading();
        GlideApp.with(mContext).load(GetUrlByScreen(item.getUrl()))
                .into(new SimpleTarget<Drawable>() {
                    @Override
                    public void onResourceReady(Drawable resource,
                                                Transition<? super Drawable> glideAnimation) {
                        mDetailImage.setImageDrawable(resource);
                        baseView.hideLoading();
                    }
                });
        container.addView(viewLayout);
        container.addView(img, LinearLayout.LayoutParams.MATCH_PARENT,
                LinearLayout.LayoutParams.MATCH_PARENT);
        return viewLayout;
    }

    private String GetUrlByScreen(String urlToChange) {
        return ViewModel.Current.getUrlByScreenSize(urlToChange);
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        ((ViewPager) container).removeView((RelativeLayout) object);
    }

}

File: ui-detail/src/main/java/com/sfaxdroid/detail/DetailPagerAdapter.kt
package com.sfaxdroid.detail

import android.content.Context
import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.RelativeLayout
import com.bumptech.glide.Glide
import com.bumptech.glide.request.target.SimpleTarget
import com.bumptech.glide.request.transition.Transition
import com.sfaxdroid.bases.BaseView
import com.sfaxdroid.base.Utils.Companion.getUrlByScreenSize
import com.sfaxdroid.base.WallpaperObject
import com.sfaxdroid.detail.utils.TouchImageView
import java.util.*

class DetailPagerAdapter(
    private val context: Context,
    private val resourceId: Int,
    wallpaperList: ArrayList<WallpaperObject?>?,
    private val baseView: BaseView
) : BasePagerAdapter<WallpaperObject?>(wallpaperList) {

    override fun isViewFromObject(
        view: View,
        `object`: Any
    ): Boolean {
        return view === `object`
    }

    override fun instantiateItem(
        container: ViewGroup,
        position: Int
    ): Any {
        val inflater = context
            .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val viewLayout = inflater.inflate(resourceId, container, false)

        val detailImage: TouchImageView = viewLayout
            .findViewById(R.id.detailImage)
        baseView.showLoading()
        Glide.with(context).load(getUrlByScreen(getItem(position)?.url ?: "", context))
            .into(object : SimpleTarget<Drawable?>() {
                override fun onResourceReady(
                    resource: Drawable,
                    transition: Transition<in Drawable?>?
                ) {
                    detailImage.setImageDrawable(resource)
                    baseView.hideLoading()
                }
            })
        container.addView(viewLayout)
        container.addView(
            TouchImageView(context), LinearLayout.LayoutParams.MATCH_PARENT,
            LinearLayout.LayoutParams.MATCH_PARENT
        )

        return viewLayout
    }

    private fun getUrlByScreen(
        urlToChange: String,
        context: Context
    ) = getUrlByScreenSize(urlToChange, context)

    override fun destroyItem(
        container: ViewGroup,
        position: Int,
        `object`: Any
    ) {
        container.removeView(`object` as RelativeLayout)
    }

}